api/controllers/SearchController.js

```diff
@@ -498,41 +575,29 @@ module.exports = {
         size: perPage,
         fields: ['package_name', 'version', 'name']
       }
-    }).then(function(response) {
-       //return res.json(response);
-      var hits = response.hits.hits.map(function(hit) {
-        var fields = {};
-        if (hit._type === 'package_version') {
-          fields.package_name = hit.fields.package_name[0];
-          fields.version = hit.fields.version[0];
-        } else if (hit._type === 'topic') {
+    }).then(function(result){
+      var functions  = [];
+      result.hits.hits.forEach(function(hit) {
+          var fields = {};
           var inner_hits_fields = hit.inner_hits.package_version.hits.hits[0].fields;
           fields.package_name = inner_hits_fields.package_name[0];
           fields.version = inner_hits_fields.version[0];
           fields.name = hit.fields.name[0];
-        }
-        return {
-          fields: fields,
-          type: hit._type,
-          score: hit._score,
-          highlight: hit.highlight
-        };
-      });
-      return res.ok({
-        total: numeral(response.hits.total).format('0,0'),
-        hits: hits,
-        perPage: perPage,
-        currentPage: page,
-        prevPageUrl: req.path + '?' + querystring.stringify(prevPageQuery),
-        nextPageUrl: req.path + '?' + querystring.stringify(nextPageQuery),
-        striptags: striptags,
```

Comment:
striptags was important because the different fields in topics can contains html tags, but the highlighted parts only contains a part of a field, so it is possible that a html tag is open but not close, that's why we all tags except `<mark>`. Before we were passing the striptags function to ejs templating system. Now you need to strip them before sending the response